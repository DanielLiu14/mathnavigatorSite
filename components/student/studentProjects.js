'use strict';
require('./studentProjects.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import {
  SiteDescription,
  Projects
} from './dataProjects.js';
const classnames = require('classnames');
const srcBlank = require('../../assets/blank.png');

const STUDENTS_DOMAIN = "https://www.students.andymathnavigator.com";

export class StudentProjectsPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: SiteDescription,
      projects: Projects || []
    }
  }

  componentDidMount() {
		if (process.env.NODE_ENV === 'production') {
      mixpanel.track("student-projects");
    }
  }

	render() {
    var sections;
    if (Projects && Projects.length > 0) {
      sections = Projects.map((project, index) =>
        <Section key={index} section={project}/>
      );
    } else {
      sections = <h3>Content Coming Soon</h3>
    }

		return (
      <div id="view-students-projects">
        <div id="view-students-projects-container">
          <h1>Student Website Projects</h1>
          <p>{this.state.description}</p>
          {sections}
        </div>
      </div>
		);
	}
}

class Section extends React.Component {
  render() {
    const section = this.props.section;
    const items = section.projects.map((project, index) => (
      <li key={index}>
        <ProjectCard project={project}/>
      </li>
    ));

    return (
      <div className="section">
        <h3>{section.sectionTitle}</h3>
        <ul>
          {items}
        </ul>
      </div>
    );
  }
}

class ProjectCard extends React.Component {
  render() {
    const project = this.props.project;
    return (
      <div className="project-card">
        <div className="img-container">
          <div className="overlay"></div>
          <OverlayContent project={project}/>
          <img src={project.imgSrc || srcBlank}/>
        </div>
        <div className="info">
          <h4>{project.student1}</h4>
          <h4>{project.title}</h4>
          <div>{project.school}, {project.grade}</div>
        </div>
      </div>
    );
  }
}

class OverlayContent extends React.Component {
  render() {
    const project = this.props.project;
    const content = createContent(project);
    return (
      <div>
        {content}
      </div>
    );
  }
}

function createContent(project) {
  const fullUrl = STUDENTS_DOMAIN + project.url;
  if (project.url) {
    return (
      <div className="card-content">
        <h4>{project.title}</h4>
        <p>{project.description}</p>
        <a href={fullUrl} target="_blank">
          <button>View</button>
        </a>
      </div>
    );
  } else {
    return (
      <div className="card-content">
        <h4>Coming Soon</h4>
      </div>
    );
  }
}
