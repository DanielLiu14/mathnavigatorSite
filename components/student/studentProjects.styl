@import "../static.styl"

#view-students-projects
  width: containerWidthPct
  max-width: containerWidthPx2
  height: 320px
  margin: spacing12x auto spacing16x

  & h1, h3
    line-height: lineHeight2
    font-family: fontFamilyRockwell
    font-size: fontSize2
    color: colorPrimary
    text-align: center
