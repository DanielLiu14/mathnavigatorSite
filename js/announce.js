'use strict';
require('./../styl/announce.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { getList } from './repos/announceRepo.js';

export class AnnouncePage extends React.Component {
	render() {
		const announcements = getList();

		const cards = announcements.map((obj, index) =>
			<AnnounceCard key={index} announcement={obj} />
		);

		return (
      <div id="view-announce">
        <div id="view-announce-container">
          <h1>Announcements</h1>
					{cards}
        </div>
      </div>
		);
	}
}

class AnnounceCard extends React.Component {
	render() {
		const announcement = this.props.announcement;
		return (
			<div className="announce-card">
				<h3>{announcement.date}</h3>
				<p>{announcement.message}</p>
				<div className="card-author">~ {announcement.author}</div>
				<div></div>
			</div>
		);
	}
}
