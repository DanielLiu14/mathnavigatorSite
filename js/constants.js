'use strict';
var _ = require('lodash/core');

export const NavLinks = [
  { id: "home", name: "Home", url: "/" },
  { id: "announce", name: "Announcements", url: "/announcements" },
  { id: "programs", name: "Programs", url: "/programs" },
  { id: "contact", name: "Contact", url: "/contact" }
];

export function getNav(id) {
  return _.find(NavLinks, {id: id});
}

export function isPathAt(url) {
  var path = location.hash;
  if (url == getNav("home").url) {
    return path == '#/';
  } else {
    return path.indexOf(url) >= 0;
  }
}
