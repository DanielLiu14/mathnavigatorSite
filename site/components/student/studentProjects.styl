@import "../static.styl"

#view-students-projects
  width: containerWidthPct
  max-width: containerWidthPx2
  margin: spacing12x auto spacing16x

  & h1, h3
    line-height: lineHeight2
    font-family: fontFamilyRockwell
    font-size: fontSize2
    color: colorPrimary
    text-align: center

  & p
    text-align: center

  & .section
    & h3
      line-height: propUnit2x
      border-bottom: 1px solid colorGrayLight

    & ul
      margin-top: spacing4x

    & li
      margin-bottom: spacing8x


#view-students-projects .project-card

  & .img-container
    position: relative
    width: 440px
    height: 330px
    margin: 0 auto

    &:hover
      & .card-content
        display: block

      & .overlay
        opacity: 0.8

    & .overlay
      position: absolute
      top: 0
      left: 0
      width: 100%
      height: 100%
      background-color: colorBlackDark
      opacity: 0.0

  & img
    display: block
    width: 100%
    height: 100%

  & .info
    margin-top: spacing2x
    line-height: lineHeight0
    font-size: fontSize0
    text-align: center

    & h4
      line-height: lineHeight1
      font-size: fontSize1
      color: colorPrimary

#view-students-projects .project-card .card-content
  display: none
  position: absolute
  top: 50%
  left: 50%
  transform: translate(-50%, -50%)
  width: 100%
  padding: spacing2x spacing4x
  box-sizing: border-box

  & h4
    margin-bottom: spacing2x
    color: colorWhite
    text-align: center

  & p
    margin: spacing2x 0
    color: colorWhite
    text-align: center

  & a
    position: absolute
    left: 50%
    transform: translateX(-50%)
