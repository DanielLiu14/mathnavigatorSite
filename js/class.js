'use strict';
require('./../styl/class.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { ErrorPage } from './error.js';
const classNames = require('classnames');
import {
  getLocation,
  getPrereqs,
  getProgramClass,
  getProgramByIds,
  getSessions
} from './repos/mainRepo.js';

function generatePrereqs(prereqIds) {
  var prereqsPrograms = getProgramByIds(prereqIds);
  var prereqsNames = prereqsPrograms.map(function(program) {
    return program ? program.title : null;
  });
  return prereqsNames.join(", ");
}

function generateLocation(locationObj) {
  return (
    <div className="class-lines">
      {locationObj.name}<br/>
      {locationObj.address1}<br/>
      {locationObj.address2}<br/>
      {locationObj.address3}
    </div>
  );
}

function generateTimes(classObj) {
  const times = classObj.times.map((time, index) =>
    <div key={index}>{time}</div>
  );
  return (
    <div className="class-lines">
      {times}
      <div>
        First session: {classObj.startDate}<br/>
        Last session: {classObj.endDate}
      </div>
    </div>
  );
}

function generatePricing(pricePerSession, numSessions) {
  let line1;
  let line2;
  let line3;
  if (numSessions > 0) {
    line1 = <div>{"Number of sessions: " + numSessions}</div>;
    line3 = <div>{"Total: $" + numSessions * pricePerSession}</div>;
  } else {

  }
  line2 = <div>{"Price per session: $" + pricePerSession}</div>;
  return (
    <div className="class-lines">
      {line1}
      {line2}
      {line3}
    </div>
  );
}

function generateSchedules(sessions) {
  if (!sessions || sessions.length == 0) {
    return (
      <div className="class-lines not-avail">
        No schedule available at the moment. <br/>
        Please check again later.
      </div>
    );
  }

  var sessionIndex = 0;
  const sessionLines = sessions.map(function(session, index) {
    var text1 = "";
    var text2 = ""
    if (session.canceled) {
      text1 = "Canceled";
      text2 = "";
    } else {
      text1 = session.time;
      text2 = session.notes;
      sessionIndex++;
    }
    return (
      <SessionLine
        key = {session.key + index}
        sessionIndex = {sessionIndex}
        date = {session.date}
        canceled = {session.canceled}
        text1 = {text1}
        text2 = {text2}
      />
    );
  });
  return (
    <ul>
      {sessionLines}
    </ul>
  );
}

export class ClassPage extends React.Component {
  render () {
    var valid = true;
    const key = this.props.slug;
    const pair = getProgramClass(key);
    valid = valid && Boolean(pair);
    valid = valid && Boolean(pair.programObj);
    valid = valid && Boolean(pair.classObj);

    var content;
    if (valid) {
      content = <ClassContent classKey={key} pair={pair}/>;
    } else {
      content = <ErrorPage classDNE={key}/>;
    }

    return (
      <div> {content} </div>
    );
  }
}

class ClassContent extends React.Component {
	render() {
    const classKey = this.props.classKey;
    const pair = this.props.pair;

    // Variables
    const programObj = pair.programObj;
    const classObj = pair.classObj;
    const locationObj = getLocation(classObj.locationId);
    let sessions = getSessions(classKey);
    sessions = sessions ? sessions : [];
    var sessionCounter = 0;
    sessions.forEach(function(session) {
      if (!session.canceled) {
        sessionCounter++;
      }
    });
    var programPrereqs = getPrereqs(programObj.programId);
    const prereqIds = programPrereqs ? programPrereqs.requiredProgramIds : [];

    // Components
    const classFullName = programObj.title + " " + classObj.className;
    const prereqs = generatePrereqs(prereqIds);
    let prereqsLine;
    if (prereqs) {
      prereqsLine = <div>{"Prequirements: " + prereqs}<br/></div>;
    }
    const textLocation = generateLocation(locationObj);
    const textTimes = generateTimes(classObj);
    const textPricing = generatePricing(classObj.pricePerSession, sessionCounter);
    const schedules = generateSchedules(sessions);

		return (
      <div id="view-class">
        <div id="view-class-container">
          <h1>
            <Link to="/programs">Programs</Link> > {classFullName}
          </h1>

          <div id="class-info-container">
            <div className="class-info-1">
              Grades: {programObj.grade1} - {programObj.grade2}<br/>
              {prereqsLine}
              {programObj.description}
            </div>

            <div className="class-info-2">
              <b>Location</b>
              {textLocation}
              <b>Times</b>
              {textTimes}
              <b>Pricing</b>
              {textPricing}
              <Link to={"/contact"}>
                <button className="inverted">Register</button>
              </Link>
            </div>
          </div>

          <div id="view-schedule">
            <b>Schedule</b>
            {schedules}
          </div>

          <div id="view-questions">
            Questions? <Link to="/contact">Contact Us</Link>
          </div>
          <Link to="/programs">
            <button className="inverted">&#60; More Programs</button>
          </Link>
        </div>
      </div>
		);
	}
}

class SessionLine extends React.Component {
  render() {
    var sessionIndex = this.props.sessionIndex;
    var date = this.props.date;
    var canceled = this.props.canceled;
    var text1 = this.props.text1;
    var text2 = this.props.text2;

    const classText1 = classNames("", {
      alert: canceled
    });
    return (
      <li>
        <div className="line-left">
          <div className="line-index">{canceled ? "" : sessionIndex}</div>
          <div className="line-date">{date}</div>
        </div>
        <div className="line-right">
          <div className={classText1}>{text1}</div>
          <div className="class-line alert">{text2}</div>
        </div>
      </li>
    );
  }
}
