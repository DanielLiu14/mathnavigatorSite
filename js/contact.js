'use strict';
require('./../styl/contact.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import { ContactInputForm } from './contactInputForm.js';
const classnames = require('classnames');


export class ContactPage extends React.Component {
	componentDidMount() {
	  window.scrollTo(0, 0);
	}

	render() {
		return (
      <div id="view-contact">
        <div id="view-contact-container">
          <h1>Contact Us</h1>
					<ContactInputForm/>
        </div>
      </div>
		);
	}
}
