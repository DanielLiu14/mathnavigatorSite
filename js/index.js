'use strict';
require('../styl/app.styl');
import React from 'react';
import ReactDOM from 'react-dom';
import {
  HashRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import { Header } from './header.js';
import { HomePage } from './home.js';
import { AnnouncePage } from './announce.js';
import { ProgramsPage } from './programs.js';
import { ContactPage } from './contact.js';
import { Footer } from './footer.js';
import { ClassPage } from './class.js';
import { ErrorPage } from './error.js';

const Home = () => <HomePage/>;
const Announce = () => <AnnouncePage/>;
const Programs = () => <ProgramsPage/>;
const Contact = () => <ContactPage/>;
const ClassPageWithSlug = ({match}) => <ClassPage slug={match.params.slug}/>;
const Error = () => <ErrorPage/>;

class MainContainer extends React.Component {
	render() {
		return (
      <Router onUpdate={() => window.scrollTo(0, 0)}>
        <div>
          <Header/>
          <Switch>
            <Route path="/" exact component={Home}/>
            <Route path="/announcements" component={Announce}/>
            <Route path="/programs" component={Programs}/>
            <Route path="/contact" component={Contact}/>
            <Route path="/class/:slug" component={ClassPageWithSlug}/>
            <Route path="/" component={Error}/>
          </Switch>
          <Footer/>
        </div>
      </Router>
		);
	}
}

ReactDOM.render(
  <MainContainer/>,
  document.getElementById('root')
);
