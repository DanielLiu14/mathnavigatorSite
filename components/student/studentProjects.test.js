import React from "react";
import Enzyme, { shallow } from "enzyme";
import { StudentProjectsPage } from "./studentProjects.js";
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe("test", () => {
  const component = shallow(<StudentProjectsPage/>);

  test("renders", () => {
    expect(component.exists()).toBe(true);
    expect(component.find("h1").text()).toBe("Student Website Projects");
  });

  test("snapshot", () => {
    const tree = renderer.create(
      <Router>
        <StudentProjectsPage/>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
