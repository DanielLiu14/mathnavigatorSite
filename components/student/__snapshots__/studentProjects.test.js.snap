// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  id="view-students-projects"
>
  <div
    id="view-students-projects-container"
  >
    <h1>
      Student Website Projects
    </h1>
    <p>
      As the final project of our Website Development Program, students create a personal website using the skills they've learned from our sessions. These websites can be for personal portfolios and hobbies or even for their school clubs.
    </p>
    <div
      className="section"
    >
      <h3>
        Summer 2019
      </h3>
      <ul>
        <li>
          <div
            className="project-card"
          >
            <div
              className="img-container"
            >
              <div
                className="overlay"
              />
              <h4>
                Coming Soon
              </h4>
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="info"
            >
              <h4>
                Jessica
              </h4>
              <h4>
                Personal Portfolio
              </h4>
              <div>
                Montgomery Blair H.S.
                , 
                12th Grade
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            className="project-card"
          >
            <div
              className="img-container"
            >
              <div
                className="overlay"
              />
              <h4>
                Coming Soon
              </h4>
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="info"
            >
              <h4>
                Daniel
              </h4>
              <h4>
                Alpha Sirius
              </h4>
              <div>
                Thomas Wootton H.S.
                , 
                10th Grade
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            className="project-card"
          >
            <div
              className="img-container"
            >
              <div
                className="overlay"
              />
              <h4>
                Coming Soon
              </h4>
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="info"
            >
              <h4>
                Richard
              </h4>
              <h4>
                Whitman Math Team
              </h4>
              <div>
                Walt Whitman H.S.
                , 
                12th Grade
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            className="project-card"
          >
            <div
              className="img-container"
            >
              <div
                className="overlay"
              />
              <h4>
                Coming Soon
              </h4>
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="info"
            >
              <h4>
                Aries
              </h4>
              <h4>
                Military Planes
              </h4>
              <div>
                Richard Montgomery H.S.
                , 
                11th Grade
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            className="project-card"
          >
            <div
              className="img-container"
            >
              <div
                className="overlay"
              />
              <h4>
                Coming Soon
              </h4>
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="info"
            >
              <h4>
                Noah
              </h4>
              <h4>
                Personal Portfolio
              </h4>
              <div>
                Clarksburg H.S.
                , 
                9th Grade
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            className="project-card"
          >
            <div
              className="img-container"
            >
              <div
                className="overlay"
              />
              <h4>
                Coming Soon
              </h4>
              <img
                src="test-file-stub"
              />
            </div>
            <div
              className="info"
            >
              <h4>
                Sean
              </h4>
              <h4>
                German Jet Fighters
              </h4>
              <div>
                Takoma Park M.S.
                , 
                8th Grade
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
